activerecord/lib/active_record/integration.rb

```diff
@@ -96,7 +96,11 @@ def cache_key(*timestamp_names)
     # Note, this method will return nil if ActiveRecord::Base.cache_versioning is set to
     # +false+ (which it is by default until Rails 6.0).
     def cache_version
-      if cache_versioning && timestamp = try(:updated_at)
+      return unless cache_versioning
+
+      if timestamp = try(:updated_at_before_type_cast)
```

Comment:
@schneems might be able to squeeze a bit more out of it by replacing `try` with `respond_to?`.
https://gist.github.com/ohaddahan/f54f9f891aa2bf6c4dc83c00e4936f86

`respond_to?` on an attribute that exists 463K vs. `try` 362K , that's 27% increase.
On a non existing attribute `respond_to?` and `try` are within error maring at 350K.

Reply:
```
if respond_to?(:updated_at_before_type_cast)
    updated_at_before_type_cast
  elsif respond_to?(:updated_at)
    updated_at.utc.to_s(:usec)
  end
```

Reply:
Please avoid commenting on open PRs with "have you tried X?" unless you have benchmark numbers to back it up

Reply:
You mean like the gist link I posted? @sgrif

Reply:
@ohaddahan We generally don't accept microbenchmarks, and this PR is discussing the context of a full request/response cycle.

Reply:
We talked about this offline. Thanks for the comment! Though I want to keep this PR to one change at a time. I think we would need to show that the above change affected the total time of execution for a request in a measurable way.

Reply:
@sgrif in the case where `update_at_before_type_cast` exist, we get a 13% improvement.
https://gist.github.com/ohaddahan/2579c1472be753a17a6a310bc5f83c8a
On the `else` case the call to `utc.to_s(:usec)` is too dominate to get significant benefit from the change to `respond_to?`.
Keep in mind that I didn't run the testing suite, so I'm assuming we don't need to propagate down stream, which is the only reason to use `try` over `respond_to?` in this case.
Also, I'm using the latest `try` from your latest PR, but if you think switching one method isn't worth 13% increase, it's your call.

Reply:
@schneems didn't meant to insert it into this PR, just a heads up to be considered as an additional PR.